### PR TITLE
Use "provided" hadoop dependencies

### LINKED ISSUE #4399: Use "provided" hadoop-* dependencies in Spark metadata client
hadoop-* dependencies are not well-shaded.  By including them in our
assembly we make users' lives harder.  Make them "provided": they _will_ be
provided in any Spark installation that is in actual use (and if not, users
should include e.g. hadoop-aws explicitly).

### PR DESCRIPTION
**Pull request by arielshaqed**
Fixes #4399.

Tested by manually including hadoop-aws packages in Spark.

### CODE REVIEW

File: docs/reference/garbage-collection.md [hunk 1]
```diff
@@ -158,6 +158,7 @@ You should specify the Uber-jar path instead of `<APPLICATION-JAR-PATH>` and run
 
   ```bash
 spark-submit --class io.treeverse.clients.GarbageCollector \
+  --packages org.apache.hadoop:hadoop-aws:2.7.7 \
```
Comment (talSofer) [inline docs/reference/garbage-collection.md:161]: @arielshaqed is this the right hadoop-aws version for all three Spark clients?

**Review by Jonathan-Rosenberg — APPROVED** [review]

File: docs/reference/garbage-collection.md (hunk 1 above)
Comment (arielshaqed) [reply docs/reference/garbage-collection.md:161]: Not really, of course.  Spark 3 with Hadoop 3 uses some 3.x.y.  E.g. the 312-hadoop3 build is _compiled_ against Hadoop 3.2.1.

Good catch, fixed this version!  But _none_ of the guide really explains Hadoop 2 vs. Hadoop 3 very well.

**Review by arielshaqed — COMMENTED** [review]
Thanks!

PTAL...

**Review by talSofer — APPROVED** [review]
Thanks! can you please update CHANGELOG to include this change before committing?

**Pull request merged**